spec/services/coronavirus_pages/timeline_entry_builder_spec.rb

```diff
@@ -0,0 +1,67 @@
+require "rails_helper"
+
+RSpec.describe CoronavirusPages::TimelineEntryBuilder do
+  let(:fixture_path) { Rails.root.join "spec/fixtures/coronavirus_landing_page.yml" }
+  let(:github_content) { YAML.safe_load(File.read(fixture_path)) }
+
+  let(:coronavirus_page) { create(:coronavirus_page, :landing) }
+
+  before do
+    stub_request(:get, Regexp.new(coronavirus_page.raw_content_url))
+      .to_return(status: 200, body: github_content.to_json)
+  end
+
+  it "creates new timeline entries from the yaml file in reverse order" do
+    described_class.new.create_timeline_entries
+
+    expect(coronavirus_page.timeline_entries.count).to eq(2)
+
+    first_entry = coronavirus_page.timeline_entries.first
+
+    expect(first_entry.heading).to eq(github_entry_list.second["heading"])
+    expect(first_entry.content).to eq(github_entry_list.second["paragraph"])
+  end
+
+  it "removes existing timeline entries before creating new ones" do
+    create(:timeline_entry, coronavirus_page: coronavirus_page)
+    create(:timeline_entry, coronavirus_page: coronavirus_page)
+    create(:timeline_entry, coronavirus_page: coronavirus_page)
+
+    described_class.new.create_timeline_entries
+
+    expect(coronavirus_page.timeline_entries.count).to eq(2)
+  end
+
+  it "doesn't remove existing timeline entries if the YAML file is empty" do
+    described_class.new.create_timeline_entries
+
+    github_content = YAML.safe_load(File.read(fixture_path))
+    github_content["content"]["timeline"]["list"].clear
+
+    stub_request(:get, Regexp.new(coronavirus_page.raw_content_url))
+      .to_return(status: 200, body: github_content.to_json)
+
+    described_class.new.create_timeline_entries
+
+    expect(coronavirus_page.timeline_entries.count).to be(2)
+  end
+
+  it "keeps the timeline entries in the right order" do
+    new_data = { "heading" => "heading", "paragraph" => "paragraph" }
+
+    github_content = YAML.safe_load(File.read(fixture_path))
+    github_content.dig("content", "timeline", "list").unshift(new_data)
+
+    stub_request(:get, Regexp.new(coronavirus_page.raw_content_url))
+      .to_return(status: 200, body: github_content.to_json)
+
+    described_class.new.create_timeline_entries
+
+    expect(coronavirus_page.timeline_entries.last.heading).to eq("heading")
+    expect(coronavirus_page.timeline_entries.last.position).to eq(1)
+  end
+
+  def github_entry_list
```

Comment:
Minor - there's a couple of other places in the code that could use this method (lines 39 and 53)

Reply:
Ah nice spot, should have seen this. I've been able to use it on 39 but because `github_content` has been redefined on line 52 trying to use it on 53 breaks things.

Reply:
Oh sorry - hadn't quite appreciated that it was using the let (not sure what I was thinking) I imagine like 39 also does something undesirable as that'll have a different `github_content` in scope to this (isn't variable scope fun?)

Reply:
oh blimey yes good point, I'll undo that one too! Changed things which made the tests pass but for all the wrong reasons! All this and it'll be the version you reviewed that actually gets merged :D